src/main/java/edu/ucsb/cs156/happiercows/controllers/CoursesController.java

```diff
@@ -69,4 +69,37 @@ public Courses postCourses(
 
         return savedCourse;
     }
-}
+
+    @Operation(summary = "Update a course")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PutMapping("")
+    public Courses updateCourses(
+               @Parameter(name="id") @RequestParam Long id,
+            @RequestBody @Valid Courses incoming) {
+
+        Courses course1 = coursesRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(Courses.class, id));
+
+       course1.setName(incoming.getName());
+       course1.setSchool(incoming.getSchool());
+       course1.setTerm(incoming.getTerm());
+       course1.setStartDate(incoming.getStartDate());
+       course1.setEndDate(incoming.getEndDate());
+
+        coursesRepository.save(course1);
+
+        return course1;
+    
+    }
+    @Operation(summary = "Delete a course")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @DeleteMapping("")
+    public Object deleteCourses(
+            @Parameter(description = "ID of the Course to delete") @RequestParam Long id) {
+        Courses course = coursesRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(Courses.class, id));
+
+        coursesRepository.delete(course);
+        return genericMessage(String.format("Course with id %s deleted", id));
+    }
```

Comment:
When deleting a course, you also need to first delete all students that have that course id.

Otherwise you end up with student in the students table that have invalid course id's